compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -186,9 +186,9 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
         let concrete = infcx.const_eval_resolve(param_env, uv.expand(), Some(span));
         match concrete {
             Err(ErrorHandled::TooGeneric) => {
-                Err(NotConstEvaluatable::Error(infcx.tcx.sess.delay_span_bug(
+                Err(NotConstEvaluatable::Error(infcx.tcx.sess.span_err(
                     span,
-                    format!("Missing value for constant, but no error reported?"),
+                    format!("unable to use constant with a hidden value in the type system"),
```

Comment:
What's a "constant with a hidden value"?

Reply:
that error message is bad but i wasn't able to think of a better one.

This can be hit in two cases:
- when encountering an opaque type during ctfe
- when encountering a default associated item (with `feature(specialization)`)

we should definitely change that message before merging this PR but I am not yet sure about the best way to do so